disnake/components.py

```diff
@@ -474,12 +476,17 @@ def to_dict(self) -> TextInputPayload:
         return payload
 
 
-def _component_factory(data: ComponentPayload) -> Component:
+MessageComponent = Union[Button, SelectMenu]
+ModalComponent = TextInput
+T = TypeVar("T", bound=Component)
+
+
+def _component_factory(data: ComponentPayload, *, type: Type[T] = Component) -> T:
```

Comment:
There should be a comment mentioning that the `type` parameter is purely for typing and doesn't affect runtime behavior